metricbeat/module/kafka/consumergroup/query.go

```diff
@@ -38,51 +39,55 @@ func fetchGroupInfo(
 
 	debugf("known consumer groups: ", groups)
 
+	assignments, err := fetchGroupAssignments(b, groups)
+	if err != nil {
+		logp.Err("failed to fetch kafka group assignments: %v", err)
+		return err
+	}
+	if len(assignments) == 0 {
+		return nil
+	}
+
 	wg := sync.WaitGroup{}
-	results := make(chan result, len(groups))
-	for _, group := range groups {
-		group := group
+	results := make(chan result, len(assignments))
```

Comment:
As the loop collecting data does not wait for wg.Wait anymore, but through all results, the wg is not really required anymore. We can even introduce an unbuffered channel `make(chan result)` now.

Reply:
fixed